src/containers/WorkspaceView.tsx

```diff
@@ -1,50 +1,184 @@
-import { Button } from '@material-ui/core';
-import React, { FunctionComponent, useCallback, useState } from 'react';
-import Splitter from '../extensions/common/Splitter';
-import { Recording, RecordingInfo } from '../reducers/recordings';
+import React, { FunctionComponent, useMemo } from 'react';
+import { useHistory, useLocation } from 'react-router';
+import { WorkspaceInfo } from '../AppContainer';
+import { Plugins } from '../extensions/extensionInterface';
+import { Recording } from '../reducers/recordings';
 import { Sorting } from '../reducers/sortings';
-import { WorkspaceInfo } from '../reducers/workspaceInfo';
-import ImportRecordingsInstructions from './ImportRecordingsInstructions';
-import RecordingsTable from './RecordingsTable';
+import SortingView from './SortingView';
+import WorkspaceRecordingsView from './WorkspaceRecordingsView';
+import WorkspaceRecordingView from './WorkspaceRecordingView';
 
 type Props = {
     workspaceInfo: WorkspaceInfo
     sortings: Sorting[]
     recordings: Recording[]
+    plugins: Plugins
     onDeleteRecordings: (recordingIds: string[]) => void
-    onSetRecordingInfo: (a: { recordingId: string, recordingInfo: RecordingInfo }) => void
     width: number
     height: number
 }
 
-const WorkspaceView: FunctionComponent<Props> = ({ width, height, sortings, recordings, onDeleteRecordings, onSetRecordingInfo, workspaceInfo }) => {
-    const [showImportInstructions, setShowImportInstructions] = useState(false)
-    const handleImport = useCallback(() => {
-        setShowImportInstructions(true)
-    }, [])
-    return (
-        <Splitter
-            {...{width, height}}
-            initialPosition={300}
-            positionFromRight={true}
-        >
-            <div>
-                {
-                    !showImportInstructions && (
-                        <div><Button onClick={handleImport}>Import recordings</Button></div>
-                    )
-                }
-                <RecordingsTable
-                    {...{sortings, recordings, onDeleteRecordings, onSetRecordingInfo, workspaceInfo}}
-                />
-            </div>
-            {
-                showImportInstructions && (
-                    <ImportRecordingsInstructions />
-                )
-            }
-        </Splitter>
+type Page = 'recordings' | 'recording' | 'sorting'
+const isPage = (x: string): x is Page => {
+    return ['recordings', 'recording', 'sorting'].includes(x)
+}
+
+type WorkspaceRecordingsRoute = {
+    page: 'recordings',
+    workspaceName: string
+}
+type WorspaceRecordingRoute = {
+    page: 'recording',
+    recordingId: string,
+    workspaceName: string
+}
+type WorspaceSortingRoute = {
+    page: 'sorting',
+    recordingId: string,
+    sortingId: string,
+    workspaceName: string
+}
+type WorkspaceRoute = WorkspaceRecordingsRoute | WorspaceRecordingRoute | WorspaceSortingRoute
+type GotoRecordingsPageAction = {
```

Comment:
Three different actions for navigating to the 3 pages: GotoRecordingsPageAction, GotoRecordingPageAction, GotoSortingPageAction